[GLib] Update version macros

TingPing opened the pull request:
#### 0f35cb3d2cfd7652354c6355810af3098bac4d1b
<pre>
[GLib] Update version macros
<a href="https://bugs.webkit.org/show_bug.cgi?id=257504">https://bugs.webkit.org/show_bug.cgi?id=257504</a>

Reviewed by Michael Catanzaro.

This updates the macros we define to match the actual version
we require and use.

GLIB_VERSION_MIN_REQUIRED should be 2_70, however we still use
deprecated APIs in a few places that need to be updated.

* Source/WTF/wtf/Platform.h:

Canonical link: <a href="https://commits.webkit.org/293200@main">https://commits.webkit.org/293200@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c9edfadce50fb3f883f9d107eb9f1f49866c568f

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/98135 "4 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/17766 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/7993 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/103251 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/48664 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/18058 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/26217 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/74723 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/31901 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/101139 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/13687 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/88679 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/55083 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/13471 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/6613 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/48106 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/90816 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/83448 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/6693 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/105628 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/96761 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/25221 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/18377 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/83716 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/25594 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/84860 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/83170 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/27799 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/5488 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/18861 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/15907 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/25180 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/30354 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/120382 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/25000 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/33727 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/28316 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/26575 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro reviewed (commented):
Hm, the `GLIB_VERSION_MAX_ALLOWED` is of limited usefulness. I think it's not currently used in order to allow conditionally using newer APIs behind preprocessor guards, which this will prevent. Adding it is useful to prevent accidental use of newer APIs, but that only actually works if we set it to `GLIB_VERSION_2_56`, which we can't do because we do want to conditionally use newer stuff if available.

With what you have here, it's still possible to accidentally depend on APIs added between 2.56 and 2.68, but no longer possible to accidentally depend on anything newer than that. (It's also no longer possible to _intentionally_ depend on anything newer than that, but hopefully anybody trying to do so would grep for `GLIB_VERSION_MAX_ALLOWED` and find it here.)

So... whatever. I'm not sure it's worth the benefit. Dunno.

Of course we should update `GLIB_VERSION_MIN_REQUIRED` though.

mcatanzaro reviewed (approved):
Bonus points if you can figure out how to avoid the style checker complaint. (Looks like it wants you to move this to a different PlatformSomething.h? @donny-dont?)

mcatanzaro reviewed (approved):
```
ERROR: Source/WTF/wtf/Platform.h:88:  Misplaced OS version check. Please use a named macro in one of headers in the wtf/Platform.h suite of files or an appropriate internal file.  [build/version_check] [5]
ERROR: Source/WTF/wtf/Platform.h:89:  Misplaced OS version check. Please use a named macro in one of headers in the wtf/Platform.h suite of files or an appropriate internal file.  [build/version_check] [5]
ERROR: Source/WTF/wtf/Platform.h:98:  Misplaced OS version check. Please use a named macro in one of headers in the wtf/Platform.h suite of files or an appropriate internal file.  [build/version_check] [5]
ERROR: Source/WTF/wtf/Platform.h:99:  Misplaced OS version check. Please use a named macro in one of headers in the wtf/Platform.h suite of files or an appropriate internal file.  [build/version_check] [5]
```

It's a false positive, but there's also a FIXME comment complaining that these definitions are in the wrong place, so maybe we should just move them and the soup defines below to OptionsGTK.cmake and OptionsWPE.cmake. (At some point in the future, we can work on reducing duplication between OptionsGTK.cmake and OptionsWPE.cmake.)

TingPing committed:
[GLib] Update version macros
https://bugs.webkit.org/show_bug.cgi?id=257504

Reviewed by Michael Catanzaro.

This updates the macros we define to match the actual version
we require and use.

GLIB_VERSION_MIN_REQUIRED should be 2_70, however we still use
deprecated APIs in a few places that need to be updated.

* Source/WTF/wtf/Platform.h:

Canonical link: https://commits.webkit.org/293200@main
